### DISCUSSION TITLE
[Bug]:Timepicker error conflicts with ACF

### DISCUSSION THREAD

**Discussion by ankitcena115** [code4recovery/12-step-meeting-list#1239] [Q&A] [answered]
### Contact Details

ankit.tiwari@netsolutions.com

### Website With Issue

https://uat-stg.aacle.org/

### What happened?

When I updated the 12 Step Meeting List Plugin its conflicts with the js of ACF and timepicker of ACF shoing issues, instead of timepicker popup its showing select list in weird format like (10:mm tt). Please fix this or tell me any way to fix this issue.

**Comment by joshreisner** [comment]
hi @ankitcena115 - i've installed Advanced Custom Fields locally and I'm not seeing any issues, have you used it to customize the meeting edit screen itself?

<img width="286" alt="Screenshot 2023-10-22 at 7 56 35 AM" src="https://github.com/code4recovery/12-step-meeting-list/assets/1551689/f6b07d3f-2c97-4786-af8a-9ea2b6c2aa1f">

we haven't made any changes to this part of the plugin in a long time

**Reply by joshreisner** [reply]
ah i see the time picker in TSML still works, but having TSML installed means that the timepicker in ACF no longer works…

**Reply by joshreisner** [reply]
i see a few years ago we took on some work specifically not to interfere with the ACF time picker, seems to no longer be working: https://github.com/code4recovery/12-step-meeting-list/pull/111

**Reply by joshreisner** [reply]
ok i've learned a couple things

* TSML and ACF use the same timepicker library
* TSML's version is (surprisingly) newer `1.13.15` (vs `1.6.3` in ACF)
* we [fixed this before](https://github.com/code4recovery/12-step-meeting-list/pull/111) by only loading TSML's javascript on TSML pages, but
* that change stopped working in August and it broke the plugin, so [we quickly removed it](https://github.com/code4recovery/12-step-meeting-list/commit/b6278b539f336ec25aa5f2c9ff1381751877a8cf)
* i can bring a new version of that change back
* timepickers will start working again, but (as before) not on TSML meeting screens

**Reply by joshreisner** [reply, marked as answer]
created a PR to fix the issue: https://github.com/code4recovery/12-step-meeting-list/pull/1240/files

if you want to test this fix now you can change lines 20-25 of admin_meeting.php to

```
add_action('admin_enqueue_scripts', function () {
	$screen = get_current_screen();
	if (is_object($screen) && $screen->post_type === 'tsml_meeting') {
		tsml_assets();
	}
});
```

**Reply by joshreisner** [reply]
`3.14.23` released with the fix

**Reply by ankitcena115** [reply]
Hi joshreisner,
Thank you for fixing the issue, its fixed now in 3.14.23 version.
Thank you very much !! appreciate for your quick response.
